Sources/SourceKitLSP/Swift/DoccDocumentation.swift

```
@@ -0,0 +1,156 @@
+      throw ResponseError.workspaceNotOpen(req.textDocument.uri)
+    }
+    let documentationManager = workspace.doccDocumentationManager
+    guard let position = req.position else {
+      throw ResponseError.invalidParams("A position must be provided for Swift files")
+    }
+    let snapshot = try documentManager.latestSnapshot(req.textDocument.uri)
+    var moduleName: String? = nil
+    var catalogURL: URL? = nil
+    if let target = await workspace.buildSystemManager.canonicalTarget(for: req.textDocument.uri) {
+      moduleName = await workspace.buildSystemManager.moduleName(for: target)
+      catalogURL = await workspace.buildSystemManager.doccCatalog(for: target)
+    }
+
+    // Search for the nearest documentable symbol at this location
+    let syntaxTree = await syntaxTreeManager.syntaxTree(for: snapshot)
+    guard
+      let nearestDocumentableSymbol = DocumentableSymbol.findNearestSymbol(
+        syntaxTree: syntaxTree,
+        position: snapshot.absolutePosition(of: position)
+      )
+    else {
+      throw ResponseError.requestFailed(doccDocumentationError: .noDocumentation)
+    }
+    // Retrieve the symbol graph as well as information about the symbol
+    let symbolPosition = await adjustPositionToStartOfIdentifier(
+      snapshot.position(of: nearestDocumentableSymbol.position),
+      in: snapshot
+    )
+    let (cursorInfo, _, symbolGraph) = try await cursorInfo(
+      req.textDocument.uri,
+      Range(symbolPosition),
+      includeSymbolGraph: true,
+      fallbackSettingsAfterTimeout: false
+    )
+    guard let symbolGraph,
+      let cursorInfo = cursorInfo.first,
+      let symbolUSR = cursorInfo.symbolInfo.usr
+    else {
+      throw ResponseError.internalError("Unable to retrieve symbol graph for the document")
+    }
+    // Locate the documentation extension and include it in the request if one exists
+    let markupExtensionFile = try? await findMarkupExtensionFile(
```

Any reason to use `try?` here. Should we throw the error or instead log it using `orLog` instead?

---

Now using `orLog()` here to log the error.